configuration/device-tree.md

```diff
@@ -448,9 +448,60 @@ When assigning to the `reg` property, the address portion of the parent node nam
+
+There is therefore a need for a method of tailoring an overlay to multiple platforms with differing hardware. Supporting them all in a single .dtbo file would require heavy use of hidden ("dormant") fragments and a switch to an on-demand symbol resolution mechanism so that a missing symbol that isn't needed doesn't cause a failure. A simpler solution is to add a facility to map an overlay name to one of several implementation files depending on the current platform.
+
+The overlay map, which is rolling out with the switch to Linux 5.4, is a file that gets loaded by the firmware at start of day. It is written in DTS source format - `overlay_map.dts`, compiled to `overlay_map.dtb` and stored in the overlays directory.
+
+This is an edited version of the current map file (the full version is available [here](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/arch/arm/boot/dts/overlays/overlay_map.dts)):
+
+```
+/ {
+    vc4-kms-v3d {
+        bcm2835;
+        bcm2711 = "vc4-kms-v3d-pi4";
+    };
+
+    vc4-kms-v3d-pi4 {
+        bcm2711;
+    };
+
+    uart5 {
+        bcm2711;
+    };
+
+    pi3-disable-bt {
+        renamed = "disable-bt";
+    };
+
+    lirc-rpi {
+        deprecated = "use gpio-ir";
+    };
+};
+```
+
+Each node has the name of an overlay that requires special handling. The properties of each node are either platform names or one of a small number of special directives. A platform name with no value (an empty property) indicates that the current overlay is compatible with the platform; for example, `vc4-kms-v3d` is compatible with the `bcm2835` platform. A non-empty value for a platform is the name of an alternative overlay to use in place of the requested one; asking for `vc4-kms-v3d` on BCM2711 results in `vc4-kms-v3d-pi4` being loaded instead. Any platform not included in an overlay's node is not compatible with that overlay.
+
+The second example node - `vc4-kms-v3d-pi4` - could be inferred from the content of `vc4-kms-v3d`, but that intelligence goes into the construction of the file, not its interpretation.
+
+In the event that a platform is not listed for an overlay, one of the special directives may apply:
+
+* The `renamed` directive indicates the new name of the overlay (which should be largely compatible with the original), but also logs a warning about the rename.
+
+* The `deprecated` directive contains a brief explanatory error message which will be logged after the common prefix `overlay '...' is deprecated:`.
+
+Remember: only exceptions need to be listed - the absence of a node for an overlay means that the default file should be used for all platforms.
```

Comment:
This seems like fairly important info - perhaps it should be moved to earlier?

Reply:
Ah ha! This time you really did miss it:
> Each node has the name of an overlay that requires special handling.